### ISSUE TITLE
ui: cluster sql queries and latency are only shown from the moment the last node rejoined

### ISSUE THREAD

**Issue by JorritSalverda** [cockroachdb/cockroach#15902]
**BUG REPORT**

I run a _CockroachDB_ cluster in _Google Container Engine_ on top of preemptible vms. These have a lifetime between 12 and 24 hours, we randomly set at start so the termination of the vms gets spread out avoiding a mass termination of hosts.

After running it for a while the _cluster overview_ page shows events for nodes rejoining, probably because their host got deleted and the pod is reallocated to another host. So far so good.

The top 2 graphs - SQL queries and SQL latency - however don't show data further back than when the last node rejoined, which in this case is 3 hours ago. Either it's coincidence and the retention for that type of data is just 3 hours or there's a bug that prevents the data from being visualized.

![cockroachdb service latency retention](https://cloud.githubusercontent.com/assets/136472/26025270/14753ffa-37e4-11e7-972e-1588ecdcef4c.png)

# Logs

**node 1**

```
2017-05-13 08:24:11	INFO	CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1) cli/start.go:575
2017-05-13 08:24:11	INFO	line format: [IWEF]yymmdd hh:mm:ss.uuuuuu goid file:line msg utf8=✓ util/log/clog.go:1011
2017-05-13 08:24:11	INFO	[config] arguments: [/cockroach/cockroach start --logtostderr --insecure --host cockroachdb-estafette-0.cockroachdb-estafette.estafette.svc.cluster.local --http-host 0.0.0.0 --join cockroachdb-estafette-public] util/log/clog.go:1011
2017-05-13 08:24:11	INFO	[config] binary: CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1) util/log/clog.go:1011
2017-05-13 08:24:11	INFO	[config] running on machine: cockroachdb-estafette-0 util/log/clog.go:1011
2017-05-13 08:24:11	INFO	[config] file created at: 2017/05/13 08:24:09 util/log/clog.go:1011
2017-05-13 08:24:21	INFO	starting cockroach node cli/start.go:338
```

**node 2**

```
2017-05-13 08:02:74	INFO	CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1) cli/start.go:575
2017-05-13 08:02:74	INFO	line format: [IWEF]yymmdd hh:mm:ss.uuuuuu goid file:line msg utf8=✓ util/log/clog.go:1011
2017-05-13 08:02:74	INFO	[config] arguments: [/cockroach/cockroach start --logtostderr --insecure --host cockroachdb-estafette-1.cockroachdb-estafette.estafette.svc.cluster.local --http-host 0.0.0.0 --join cockroachdb-estafette-public] util/log/clog.go:1011
2017-05-13 08:02:74	INFO	[config] binary: CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1) util/log/clog.go:1011
2017-05-13 08:02:74	INFO	[config] running on machine: cockroachdb-estafette-1 util/log/clog.go:1011
2017-05-13 08:02:74	INFO	[config] file created at: 2017/05/13 08:02:39 util/log/clog.go:1011
2017-05-13 08:02:85	INFO	starting cockroach node cli/start.go:338
```

**node 3**

```
2017-05-13 01:25:34	INFO	CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1) cli/start.go:575
2017-05-13 01:25:34	INFO	line format: [IWEF]yymmdd hh:mm:ss.uuuuuu goid file:line msg utf8=✓ util/log/clog.go:1011
2017-05-13 01:25:34	INFO	[config] arguments: [/cockroach/cockroach start --logtostderr --insecure --host cockroachdb-estafette-2.cockroachdb-estafette.estafette.svc.cluster.local --http-host 0.0.0.0 --join cockroachdb-estafette-public] util/log/clog.go:1011
2017-05-13 01:25:34	INFO	[config] binary: CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1) util/log/clog.go:1011
2017-05-13 01:25:34	INFO	[config] running on machine: cockroachdb-estafette-2 util/log/clog.go:1011
2017-05-13 01:25:34	INFO	[config] file created at: 2017/05/13 01:25:46 util/log/clog.go:1011
2017-05-13 01:25:45	INFO	starting cockroach node cli/start.go:338
```

**node 4**

```
2017-05-13 05:45:80	INFO	CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1) cli/start.go:575
2017-05-13 05:45:80	INFO	line format: [IWEF]yymmdd hh:mm:ss.uuuuuu goid file:line msg utf8=✓ util/log/clog.go:1011
2017-05-13 05:45:80	INFO	[config] arguments: [/cockroach/cockroach start --logtostderr --insecure --host cockroachdb-estafette-3.cockroachdb-estafette.estafette.svc.cluster.local --http-host 0.0.0.0 --join cockroachdb-estafette-public] util/log/clog.go:1011
2017-05-13 05:45:80	INFO	[config] binary: CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1)util/log/clog.go:1011
2017-05-13 05:45:80	INFO	[config] running on machine: cockroachdb-estafette-3 util/log/clog.go:1011
2017-05-13 05:45:80	INFO	[config] file created at: 2017/05/13 05:45:57 util/log/clog.go:1011
2017-05-13 05:45:92	INFO	starting cockroach node cli/start.go:338
```

**node 5**

```
2017-05-13 04:25:24	INFO	CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1) cli/start.go:575
2017-05-13 04:25:24	INFO	line format: [IWEF]yymmdd hh:mm:ss.uuuuuu goid file:line msg utf8=✓ util/log/clog.go:1011
2017-05-13 04:25:24	INFO	[config] arguments: [/cockroach/cockroach start --logtostderr --insecure --host cockroachdb-estafette-4.cockroachdb-estafette.estafette.svc.cluster.local --http-host 0.0.0.0 --join cockroachdb-estafette-public] util/log/clog.go:1011
2017-05-13 04:25:24	INFO	[config] binary: CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1) util/log/clog.go:1011
2017-05-13 04:25:24	INFO	[config] running on machine: cockroachdb-estafette-4 util/log/clog.go:1011
2017-05-13 04:25:24	INFO	[config] file created at: 2017/05/13 04:25:16 util/log/clog.go:1011
2017-05-13 04:25:35	INFO	starting cockroach node cli/start.go:338
```

# Observation

Top 2 graphs only show last 3 hours of data, approximately since the last node rejoined.

**Comment by JorritSalverda** [comment]
All other dashboards show data for the full range, except for a couple in the SQL dashboard. Again _queries_ and _service latency_.

**Comment by JorritSalverda** [comment]
Interestingly enough I now see the data stopping for the last hour or so, without any of the nodes being restarted, so it could well be unrelated to nodes rejoining.

![cockroachdb incomplete graphs](https://cloud.githubusercontent.com/assets/136472/26028396/c9501068-381f-11e7-8069-eb17948fd7ef.png)

**Comment by JorritSalverda** [comment]
Over the last few days I haven't seen any relation to the nodes rejoining. This cluster isn't serving any application yet. Is it possible that the **internal queries** that must be the ones shown pause for a while every now and then, so no data is visible because no actual queries are performed?

**Comment by dianasaur323** [comment]
thanks for filing this issue, @JorritSalverda. Adding @mrtracy to take a look.

**Comment by cuongdo** [comment]
@dianasaur323 please triage

**Comment by dianasaur323** [comment]
man, this totally fell off my radar. I think we can get to this bug fix in conjunction with the other graphing bug fixes we have for 1.2.

**Comment by jordanlewis** [comment]
Closing due to age.
